managed/ui/src/components/tables/BackupStorageOptions.js

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) YugaByte, Inc.
+//
+// Author: Nishant Sharma(nishant.sharma@hashedin.com)
+//
+// This file will return the set of storage type options.
+
+import React from 'react';
+import { isNonEmptyArray, isNonEmptyObject } from '../../utils/ObjectUtils';
+
+export const BackupStorageOptions = (storageConfigs) => {
+  let configTypeList = <option />;
+  const regex = /\d+(?!\.)/;
+  const optGroups =
+    storageConfigs &&
+    isNonEmptyArray(storageConfigs) &&
+    storageConfigs.reduce((val, indx) => {
+      const configType = indx.name;
+      val[configType]
+        ? val[configType].push({
+            configName: indx.configName,
+            configUUID: indx.configUUID
+          })
+        : (val[configType] = [
+            {
+              configName: indx.configName,
+              configUUID: indx.configUUID
+            }
+          ]);
```

Comment:
Don't like the logic here. Should be an if-else block. Or if you want to be space-efficient, you can do this:
```
const currentConfig = { configName: index.configName, configUUID: indx.configUUID };
val[configType] = [...val[configType] ?? [], currentConfig];
```

Reply:
Done.